doc/source/performance.rst

```diff
@@ -0,0 +1,100 @@
+.. _performance_considerations:
+
+.. currentmodule:: pandas
+
+.. ipython:: python
+   :suppress:
+
+   import pandas as pd
+   import numpy as np
+   np.random.seed(123456)
+
+******************
+Performance Considerations
+******************
+
+Overview
+--------
+Pandas, for most use cases, will be fast; however, there are a few
+anti-patterns to avoid. The following is a guide on achieving better
+performance with pandas.
```

Comment:
Maybe add a note like, "these are the low-hanging optimizations that should be considered before rewriting your code in Cython or to use Numba"